### PR TITLE
Update clap to 4.4

### PR DESCRIPTION
**Pull request by patrickelectric**
Helps #321 

It appears that it did not fix the problem, this PR address the issue using the new configuration on clap 4.4 `default_missing_value`

### CODE REVIEW

**Comment by joaoantoniocardoso** [comment]
Closes #122

**Review by joaoantoniocardoso — CHANGES_REQUESTED** [review]
image

I don't know why, but somehow it is defining the wrong possible values

File: src/cli/manager.rs [hunk 1]
```diff
@@ -40,94 +94,74 @@ pub fn init() {
...
-pub fn vehicle_ddns() -> Option<&'static str> {
-    MANAGER.as_ref().clap_matches.value_of("vehicle-ddns")
+pub fn vehicle_ddns() -> Option<String> {
+    MANAGER.as_ref().clap_matches.vehicle_ddns.clone()
 }
 
-pub fn default_settings() -> Option<&'static str> {
-    return MANAGER.as_ref().clap_matches.value_of("default-settings");
+pub fn default_settings() -> Option<custom::CustomEnvironment> {
+    return MANAGER.as_ref().clap_matches.default_settings.clone();
```
Comment (joaoantoniocardoso) [inline src/cli/manager.rs:138]:
```suggestion
    MANAGER.as_ref().clap_matches.default_settings.clone()
```

File: src/cli/manager.rs [hunk 2]
```diff
@@ -40,94 +94,74 @@ pub fn init() {
...
 }
 
 pub fn is_reset() -> bool {
-    return MANAGER.as_ref().clap_matches.is_present("reset");
+    return MANAGER.as_ref().clap_matches.reset;
 }
 
 pub fn is_tracy() -> bool {
-    return MANAGER.as_ref().clap_matches.is_present("enable-tracy");
+    return MANAGER.as_ref().clap_matches.enable_tracy;
```
Comment (joaoantoniocardoso) [inline src/cli/manager.rs:109]:
```suggestion
    MANAGER.as_ref().clap_matches.enable_tracy
```

File: src/cli/manager.rs [hunk 3]
```diff
@@ -40,94 +94,74 @@ pub fn init() {
...
-    return MANAGER.as_ref().clap_matches.is_present("verbose");
+    return MANAGER.as_ref().clap_matches.verbose;
 }
 
 pub fn is_tracing() -> bool {
-    return MANAGER
-        .as_ref()
-        .clap_matches
-        .is_present("enable-tracing-level-log-file");
+    return MANAGER.as_ref().clap_matches.enable_tracing_level_log_file;
```
Comment (joaoantoniocardoso) [inline src/cli/manager.rs:101]:
```suggestion
    MANAGER.as_ref().clap_matches.enable_tracing_level_log_file
```

File: src/cli/manager.rs [hunk 4]
```diff
@@ -40,94 +94,74 @@ pub fn init() {
 
 // Check if the verbosity parameter was used
 pub fn is_verbose() -> bool {
-    return MANAGER.as_ref().clap_matches.is_present("verbose");
+    return MANAGER.as_ref().clap_matches.verbose;
```
Comment (joaoantoniocardoso) [inline src/cli/manager.rs:97]:
```suggestion
    MANAGER.as_ref().clap_matches.verbose
```

**Review by joaoantoniocardoso — APPROVED** [review]
Well done, thanks!!

**Comment by patrickelectric** [comment]
Thanks for the review!

**Review by joaoantoniocardoso — APPROVED** [review]

**Pull request merged**